### ISSUE TITLE
Bug in install-src

### ISSUE THREAD

**Issue by cradcore** [miurahr/aqtinstall#453]
I am trying to install the source files for QtWebEngine, linux desktop 5.9.9 gcc_64. However, this is what I get:

```
cradcore@localhost: ~ $ aqt install-src -m qtwebengine -O /dev-tools linux desktop 5.9.9
aqtinstall(aqt) v2.0.2 on Python 3.8.10 [CPython GCC 9.3.0]
The packages ['qtwebengine'] were not found while parsing XML of package information!
==============================Suggested follow-up:==============================
* Please use 'aqt list-qt linux desktop --modules 5.9.9 <arch>' to show modules available.
cradcore@localhost: ~ $ aqt list-qt linux desktop --modules 5.9.9 gcc_64
qtcharts qtdatavis3d qtnetworkauth qtpurchasing qtremoteobjects qtscript qtspeech qtvirtualkeyboard qtwebengine
```

It seems to not find qtwebengine when trying to install, but correctly lists it as a module

**Comment by ddalcino** [comment]
Looks like the "Suggested follow-up" message in this case is inappropriate. The `aqt list-qt` command correctly shows output pertinent to the `install-qt` command; it is not designed to provide data relevant to `install-src/doc/example`.

See the Updates.xml file at https://download.qt.io/online/qtsdkrepository/linux_x64/desktop/qt5_599_src_doc_examples/Updates.xml: there is no source module for qtwebengine. The archive `qtwebengine-opensource-src-5.9.9.7z` is part of the `qt.qt5.599.src` PackageUpdate; that means that you can install it by running `aqt install-src linux desktop 5.9.9`.

As far as I can tell, there are no valid options that you can pass to the `-m` flag for `install-src`. Unless I have overlooked something, maybe we should remove the flag.

The `list-qt` command is technically capable of to showing data relevant to `install-src/doc/example`, by using `--extension src_doc_examples`, but it doesn't list modules and it isn't documented properly. Right now, these are the only commands that work:
```
aqt list-qt linux desktop --extension src_doc_examples --archives 5.9.9 src
aqt list-qt linux desktop --extension src_doc_examples --archives 5.9.9 doc
aqt list-qt linux desktop --extension src_doc_examples --archives 5.9.9 examples
```

### Todo:
1. Remove/replace the "Suggested follow-up" message for `install-src/doc/example` when a module cannot be located
2. Consider adding commands for `list-src/doc/example`.
3. Consider removing the `-m` flag for `aqt install-src`.
4. Remove `src_doc_examples` from the list of valid extensions for `list-qt`, since `list-qt` is not designed to use it properly.

**Comment by ddalcino** [comment]
Also worth noting, and indirectly related to this bug: `install-src/doc/example` will only work on the "desktop" target. I think there are three potential ways to fix this:
1. Remove the `target` parameter from the cli for `install-src/doc/example` (Probably the right solution, but I am reluctant to do this because it's a big change to the interface)
2. Ignore the `target` parameter for `install-src/doc/example`, and just fill it in with "desktop" regardless of what the user types
3. If the user puts anything into the `target` parameter that's not "desktop", give an error message that says "There are no src/doc/examples for any target other than `desktop`".

**Comment by miurahr** [comment]
I prefer 1.
> Remove the target parameter from the cli 

but 2 may be better for consistency of UI with other command argument.
> Ignore the target parameter 

Which one is better on  "no surprise for user" basis?

**Comment by ddalcino** [comment]
> I prefer 1.
> 
> > Remove the target parameter from the cli
> 
> but 2 may be better for consistency of UI with other command argument.
> 
> > Ignore the target parameter
> 
> Which one is better on "no surprise for user" basis?

I think if we released aqt v2.1.0 with option 1, we would break a lot of people's code. That would badly surprise users.

Maybe there is a way to make the `target` parameter optional for these commands, for backwards compatibility, so we can have the benefits of option 1 without breaking existing code?

I think that removing the `-m` flag for `install-src` qualifies as a bug fix rather than an interface change, because there's no way to use it without experiencing the bug that @cradcore is seeing. I expect that `-m all` might not trigger the bug, but it won't actually do anything.
